### PR TITLE
:bug: include `use` statements in generated resources

### LINKED ISSUE #2: Generated Models need to add Use statements
Currently, the generated models do not include any use statements for the classes it needs in the methods.

Modify the template to get a list of all classes we need to `use` in the model.

### PR DESCRIPTION
**Pull request by bretterer**
This adds logic to include the required `use` statements in the php classes

resolves #2

### CODE REVIEW

**Comment by bretterer** [comment]
# [Codecov](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=h1) Report
> Merging [#5](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=desc) into [develop](https://codecov.io/gh/okta/okta-sdk-php/commit/ac338b94db947ba9b8f011cc9e8e5c61d529cad2?src=pr&el=desc) will **increase** coverage by `0.11%`.
> The diff coverage is `0.61%`.

[Impacted file tree graph](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree)

```diff
@@              Coverage Diff              @@
##             develop       #5      +/-   ##
=============================================
+ Coverage      15.76%   15.87%   +0.11%     
+ Complexity       349      348       -1     
=============================================
  Files             47       47              
  Lines            996      989       -7     
=============================================
  Hits             157      157              
+ Misses           839      832       -7
```


| [Impacted Files](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree) | Coverage Δ | Complexity Δ | |
|---|---|---|---|
| [src/Users/PasswordCredential.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL1VzZXJzL1Bhc3N3b3JkQ3JlZGVudGlhbC5waHA=) | `0% <0%> (ø)` | `2 <1> (ø)` | :arrow_down: |
| [src/Apps/PublicAppInstanceVisibility.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL0FwcHMvUHVibGljQXBwSW5zdGFuY2VWaXNpYmlsaXR5LnBocA==) | `0% <0%> (ø)` | `4 <3> (ø)` | :arrow_down: |
| [src/Users/ResetPasswordToken.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL1VzZXJzL1Jlc2V0UGFzc3dvcmRUb2tlbi5waHA=) | `0% <0%> (ø)` | `1 <1> (ø)` | :arrow_down: |
| [src/Apps/PublicAppInstanceLicensing.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL0FwcHMvUHVibGljQXBwSW5zdGFuY2VMaWNlbnNpbmcucGhw) | `0% <0%> (ø)` | `1 <1> (ø)` | :arrow_down: |
| [.../Shared/GroupMembershipMediationGroupCondition.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL1NoYXJlZC9Hcm91cE1lbWJlcnNoaXBNZWRpYXRpb25Hcm91cENvbmRpdGlvbi5waHA=) | `0% <0%> (ø)` | `2 <2> (?)` | |
| [...Shared/GroupMembershipMediationPeopleCondition.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL1NoYXJlZC9Hcm91cE1lbWJlcnNoaXBNZWRpYXRpb25QZW9wbGVDb25kaXRpb24ucGhw) | `0% <0%> (ø)` | `4 <2> (?)` | |
| [src/Apps/PublicAppInstanceCredentials.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL0FwcHMvUHVibGljQXBwSW5zdGFuY2VDcmVkZW50aWFscy5waHA=) | `0% <0%> (ø)` | `9 <6> (ø)` | :arrow_down: |
| [src/Shared/AssignUserToGroupsMediationAction.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL1NoYXJlZC9Bc3NpZ25Vc2VyVG9Hcm91cHNNZWRpYXRpb25BY3Rpb24ucGhw) | `0% <0%> (ø)` | `1 <1> (?)` | |
| [src/Shared/Link.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL1NoYXJlZC9MaW5rLnBocA==) | `0% <0%> (ø)` | `8 <8> (?)` | |
| [src/Users/AppLink.php](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree#diff-c3JjL1VzZXJzL0FwcExpbmsucGhw) | `0% <0%> (ø)` | `10 <10> (ø)` | :arrow_down: |
| ... and [29 more](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=footer). Last update [ac338b9...a11d18c](https://codecov.io/gh/okta/okta-sdk-php/pull/5?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: openapi/generator/index.js [hunk 1]
```diff
@@ -4,14 +4,29 @@ _.mixin(require("lodash-inflection"));
...
+            return obj.commonType;
+    }
+}
+
+function getSafeType(obj) {
+    switch (obj.commonType) {
+        case 'dateTime':
+            return ``;
+        case 'object':
+            return `: ` + obj.model;
```
Comment (lboyette-okta) [inline openapi/generator/index.js]: nit: this is usually something like \`: ${obj.model}`

File: openapi/generator/index.js [hunk 2]
```diff
@@ -4,14 +4,29 @@ _.mixin(require("lodash-inflection"));
...
+        case 'hash':
+            return String.raw`: array`;
+        default:
+            return `: ` + obj.commonType;
```
Comment (lboyette-okta) [inline openapi/generator/index.js]: nit: \`: ${obj. commonType}`

File: openapi/generator/templates/collection.php.hbs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ ******************************************************************************/
+
+namespace Okta\\{{namespace}};
+
+use Okta\Resource\AbstractCollection;
+
+class Collection extends AbstractCollection
+{
+
+}
```
Comment (lboyette-okta) [inline openapi/generator/templates/collection.php.hbs]: interesting. I guess the formatter adds a newline for each of the generated collections

**Pull request merged**